index.bs

```diff
@@ -78,4 +251,313 @@ interface StorageBucket {
+
+    1. Let |bucket| be [=this=]'s [=/storage bucket=].
+
+    1. If |bucket| has been removed, [=reject=] |p| with {{InvalidStateError}}.
+
+    1. Otherwise,
+
+        1. Let |environment| be [=/this=]'s [=/relevant settings object=].
+
+        1. Let |permission| be the result of [=getting the current permission state=] with `"persistent-storage"` and |environment|.
+
+        1. If |permission| is "{{PermissionState/granted}}", then set |bucket|'s [=bucket mode=] to `"persistent"` and [=/resolve=] |p| with true.
+
+        1. Otherwise, [=/resolve=] |p| with false.
+
+1. Return |p|.
+
+</div>
+
+<div algorithm>
+
+The <dfn method for="StorageBucket">persisted()</dfn> method steps are:
+
+1. Let |p| be [=a new promise=].
+
+1. Run the following steps [=in parallel=]:
+
+    1. Let |bucket| be [=this=]'s [=/storage bucket=].
+
+    1. If |bucket| has been removed, [=reject=] |p| with an {{InvalidStateError}}.
+
+    1. Otherwise,
+
+        1. If |bucket|'s [=bucket mode=] is `"persistent"` and [=/resolve=] |p| with true.
+
+        1. Otherwise, [=/resolve=] |p| with false.
+
+1. Return |p|.
+
+</div>
+
+<h3 id="storage-bucket-quota">Quota</h3>
+
+A [=/storage bucket=] has a <dfn>bucket quota</dfn>, a number-or-null, initially null.
+Specifies the upper limit of usage which can be used by the bucket. The user agent MAY further limit
+the realized storage space.
+
+The <dfn>storage usage</dfn> of a [=/storage bucket=] is an [=implementation-defined=] rough estimate
+of the number of bytes used by all of its [=/storage bottle=]s.
+
+<div algorithm>
+
+The <dfn method for="StorageBucket">estimate()</dfn> method steps are:
+
+1. Let |p| be [=a new promise=].
+
+1. Let |environment| be [=/this=]'s [=/relevant settings object=].
+
+1. Let |shelf| be the result of running [=obtain a local storage shelf=] with |environment|.
+
+1. If |shelf| is failure, [=reject=] p with an {{UnknownError}}.
+
+1. Otherwise, run the following steps [=in parallel=]:
+
+    1. Let |bucket| be [=this=]'s [=/storage bucket=].
+
+    1. If |bucket| has been removed, [=reject=] |p| with an {{InvalidStateError}}.
+
+    1. Otherwise,
+
+        1. Let |quota| be [=storage quota=] for |shelf|.
+
+        1. Set |quota| to |bucket|'s [=bucket quota=] if it is non-null.
+
+        1. Let |usage| be [=storage usage=] for |bucket|.
+
+        1. Let |dictionary| be a new {{StorageEstimate}} dictionary whose {{StorageEstimate/usage}} member is |usage| and {{StorageEstimate/quota}} member is |quota|.
+
+        1. [=/Resolve=] |p| with |dictionary|.
+
+1. Return |p|.
+
+</div>
+
+<aside class="note">
+
+The [=bucket quota=] will be ignored if it exceeds the total amount of space available to the [=/storage shelf=], i.e. site.
+Its intended use is to keep a specific bucket from using up the entire site's storage space.
+
+</aside>
+
+<h3 id="storage-bucket-durability">Durability</h3>
+
+A [=/storage bucket=] has a <dfn>bucket durability</dfn>, a {{StorageBucketDurability}}.
```

Comment:
Should this have a `for=StorageBucket` attribute?  And maybe change this to `durability value`?  Then you can reference it with `[=StorageBucket/durability value=]`.  (I assume the extra word is to disambiguate with the method.)

Reply:
SG, done